src/fdcache_entity.cpp

```diff
@@ -1224,12 +1223,22 @@ int FdEntity::RowFlush(const char* tpath, bool force_sync)
                     return static_cast<ssize_t>(result);
                 }
             }else{
-              // no enough disk space
-              // upload all by multipart uploading
-              if(0 != (result = NoCacheLoadAndPost())){
-                  S3FS_PRN_ERR("failed to upload all area by multipart uploading(errno=%d)", result);
-                  return static_cast<ssize_t>(result);
-              }
+                // no enough disk space
+                if(nomultipart){
+                    S3FS_PRN_WARN("Not enough local storage to flush: [path=%s][fd=%d]", path.c_str(), fd);
+                    return -ENOSPC;   // No space left on device
+                }
+                if(0 != (result = NoCachePreMultipartPost())){
+                    S3FS_PRN_ERR("failed to switch multipart uploading with no cache(errno=%d)", result);
+                    return static_cast<ssize_t>(result);
```

Comment:
I think these `ssize_t` casts are not necessary.  `Load`, `NoCacheLoadAndPost`, and `NoCachePreMultipartPost` all return `int`.  I am OK merging this PR but I am concerns about the free-space handling generally.

Reply:
It is changed not to do static_cast and changed the same part.

This change is simply a bug in processing omissions, so it's a different issue than you're concerned about.
I agree about regarding the check of free space. I also feel that the problem is that the area size that must be checked differs depending on the upload mode.

Reply:
Reading your discussion and reflecting my suggested change in #1647, I'll rework the change and move the check to `NoCachePreMultipartPost()` - it's be better placed and would also enforce the minimum size for this case.

Reply:
Agreed that this improves the correctness.  Let's think some more about free space handling in future PRs.

Reply:
Thanks